app/src/main/java/com/poupa/vinylmusicplayer/dialogs/BottomSheetDialog/BottomSheetDialog.java

```diff
@@ -0,0 +1,51 @@
+package com.poupa.vinylmusicplayer.dialogs.BottomSheetDialog;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.FrameLayout;
+
+import androidx.annotation.NonNull;
+import com.google.android.material.bottomsheet.BottomSheetBehavior;
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+
+
+public abstract class BottomSheetDialog extends BottomSheetDialogFragment {
+    public static BottomSheetDialog newInstance() { return null; }
+
+    @NonNull @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+
+        com.google.android.material.bottomsheet.BottomSheetDialog
+                dialog = new com.google.android.material.bottomsheet.BottomSheetDialog(getActivity());
+
+        dialog.setOnShowListener(new DialogInterface.OnShowListener() {
+            @Override
+            public void onShow(DialogInterface dialog) {
+                com.google.android.material.bottomsheet.BottomSheetDialog
+                        d = (com.google.android.material.bottomsheet.BottomSheetDialog) dialog;
+                FrameLayout bottomSheet = d.findViewById(com.google.android.material.R.id.design_bottom_sheet);
+
+                BottomSheetBehavior behaviour = BottomSheetBehavior.from(bottomSheet);
+                behaviour.setState(BottomSheetBehavior.STATE_COLLAPSED);
+                behaviour.setDraggable(false);
+                behaviour.addBottomSheetCallback(new BottomSheetBehavior.BottomSheetCallback() {
+                    @Override public void onStateChanged(@NonNull View bottomSheet, int newState) {
+                        if (newState == BottomSheetBehavior.STATE_EXPANDED) {
+                            behaviour.setState(BottomSheetBehavior.STATE_COLLAPSED);
+                        }
+                    }
+
+                    @Override
```

Comment:
You cannot avoid having an empty overriden function?

Reply:
BottomSheetBehavior.BottomSheetCallback is abstract and need a slide declaration